src/astro_databricks/plugins/plugin.py

```diff
@@ -194,15 +297,31 @@ def get_link(
         *,
         ti_key: TaskInstanceKey | None = None,
     ) -> str:
+        ti = None
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+        task_group = get_task_group(operator)
         self.log.debug(
             "Creating link to repair all tasks for databricks job run %s",
-            operator.task_group.group_id,
+            task_group.group_id,
         )
         # Should we catch the exception here if there is no return value?
-        metadata = XCom.get_value(
-            ti_key=ti_key,
-            key="return_value",
-        )
+        try:
+            metadata = XCom.get_value(
+                ti_key=ti_key,
+                key="return_value",
+            )
+        except AttributeError:
+            if not ti:
+                raise TaskInstanceNotFound()
+            metadata = XCom.get_one(
+                task_id=ti_key.task_id,
+                dag_id=ti_key.dag_id,
+                execution_date=ti.execution_date,
+                key="return_value",
+            )
```

Comment:
This is very similar to a previous block, perhaps we should have a function to wrap this behaviour

Reply:
yes, good catch. Done so.